File: lab/python/pares/par_impar_divisivel3.py
def par_impar_divisivel3(numero):
    try:
        numero = int(numero)  # garante que seja inteiro
    except (TypeError, ValueError):
        return "Entrada inválida"

    if numero % 2 == 0:
        return "Par"
    elif numero % 3 == 0:
        return "Ímpar e divisível por 3"
    else:
        return "Ímpar"


# Testes
numeros = [2, 9, 5, "Ola"]
for numero in numeros:
    print(par_impar_divisivel3(numero))

File: lab/javascript/pares/par_impar_divisivel3.js
function parImparDivisivel3(numero) {
    // tenta converter para inteiro
    numero = Number(numero);

    if (isNaN(numero) || !Number.isInteger(numero)) {
        return "Entrada inválida";
    }

    if (numero % 2 === 0) {
        return "Par";
    } else if (numero % 3 === 0) {
        return "Ímpar e divisível por 3";
    } else {
        return "Ímpar";
    }
}

// Testes
let numeros = [2, 9, 5, "Ola"];
for (let numero of numeros) {
    console.log(parImparDivisivel3(numero));
}
